tests/system/Security/SecurityCSRFSessionRandomizeTokenTest.php

```diff
@@ -0,0 +1,263 @@
+
+        $_SESSION = [];
+        Factories::reset();
+
+        $config                 = new SecurityConfig();
+        $config->csrfProtection = Security::CSRF_PROTECTION_SESSION;
+        $config->tokenRandomize = true;
+        Factories::injectMock('config', 'Security', $config);
+
+        $this->injectSession($this->hash);
+    }
+
+    private function createSession($options = []): Session
+    {
+        $defaults = [
+            'sessionDriver'            => 'CodeIgniter\Session\Handlers\FileHandler',
+            'sessionCookieName'        => 'ci_session',
+            'sessionExpiration'        => 7200,
+            'sessionSavePath'          => null,
+            'sessionMatchIP'           => false,
+            'sessionTimeToUpdate'      => 300,
+            'sessionRegenerateDestroy' => false,
+            'cookieDomain'             => '',
+            'cookiePrefix'             => '',
+            'cookiePath'               => '/',
+            'cookieSecure'             => false,
+            'cookieSameSite'           => 'Lax',
+        ];
+
+        $config    = array_merge($defaults, $options);
+        $appConfig = new AppConfig();
+
+        foreach ($config as $key => $c) {
+            $appConfig->{$key} = $c;
+        }
+
+        $session = new MockSession(new ArrayHandler($appConfig, '127.0.0.1'), $appConfig);
+        $session->setLogger(new TestLogger(new LoggerConfig()));
+
+        return $session;
+    }
+
+    private function injectSession(string $hash): void
+    {
+        $session = $this->createSession();
+        $session->set('csrf_test_name', $hash);
+        Services::injectMock('session', $session);
+    }
+
+    public function testHashIsReadFromSession()
+    {
+        $security = new MockSecurity(new MockAppConfig());
+
+        $this->assertSame(
+            $this->randomizedToken,
+            $security->getHash()
+        );
+    }
+
+    public function testCSRFVerifyPostThrowsExceptionOnNoMatch()
+    {
+        $this->expectException(SecurityException::class);
+
+        $_SERVER['REQUEST_METHOD'] = 'POST';
+        $_POST['csrf_test_name']   = '8b9218a55906f9dcc1dc263dce7f005b';
+
+        $request = new IncomingRequest(new MockAppConfig(), new URI('http://badurl.com'), null, new UserAgent());
+
+        $security = new Security(new MockAppConfig());
+
+        $security->verify($request);
+    }
+
+    public function testCSRFVerifyPostReturnsSelfOnMatch()
+    {
+        $_SERVER['REQUEST_METHOD'] = 'POST';
+        $_POST['foo']              = 'bar';
+        $_POST['csrf_test_name']   = $this->randomizedToken;
+
+        $request = new IncomingRequest(new MockAppConfig(), new URI('http://badurl.com'), null, new UserAgent());
+
+        $security = new Security(new MockAppConfig());
+
+        $this->assertInstanceOf(Security::class, $security->verify($request));
+        $this->assertLogged('info', 'CSRF token verified.');
+        $this->assertTrue(count($_POST) === 1);
```

Comment:
use assertCount

Reply:
No CS Fixer rule for this, like the `assertTrue/False` checks?

Reply:
Rector can fix it.
I sent a PR #5320